src/mono/mono/metadata/marshal.c

```diff
@@ -3031,6 +3076,76 @@ mono_marshal_boolean_managed_conv_in_get_conv_arg_class (MonoMarshalSpec *spec,
 	return conv_arg_class;
 }
 
+static int
+mono_emit_disabled_marshal (EmitMarshalContext *m, int argnum, MonoType *t,
+	      MonoMarshalSpec *spec, int conv_arg,
+	      MonoType **conv_arg_type, MarshalAction action)
+{
+	// Some phases like MARSHAL_ACTION_PUSH need to still emit "correct" IL for the IL stack to be valid.
+	// For those phases, we won't emit the exception throw.
+	// In all other phases we'll emit the exception throw.
+	gboolean emit_exception = action == MARSHAL_ACTION_CONV_IN || action == MARSHAL_ACTION_CONV_RESULT;
+
+	if (m_type_is_byref(t)) {
+		if (emit_exception)
+			get_marshal_cb ()->emit_marshal_directive_exception (m, argnum, "Cannot marshal managed references when the built-in marshalling system is disabled.");	
```

Comment:
Should such error messages include the assembly name?

Reply:
Since the P/Invoke method will be included in the stack trace, I don't think we need to include the assembly name here.